src/ucp/core/ucp_ep.c

```diff
@@ -139,36 +149,67 @@ ucs_status_t ucp_ep_create_base(ucp_worker_h worker, const char *peer_name,
     ucs_debug("created ep %p to %s %s", ep, ucp_ep_peer_name(ep), message);
     return UCS_OK;
 
+err_free_keys:
+    ucs_free(ucp_ep_ext_gen(ep)->ids);
 err_free_ep:
     ucs_strided_alloc_put(&worker->ep_alloc, ep);
 err:
     return status;
 }
 
-ucs_status_t ucp_worker_create_ep(ucp_worker_h worker, const char *peer_name,
-                                  const char *message, ucp_ep_h *ep_p)
+static void ucp_ep_destroy_base(ucp_ep_h ep)
+{
+    ucs_free(ucp_ep_ext_gen(ep)->ids);
+    ucs_strided_alloc_put(&ep->worker->ep_alloc, ep);
+}
```

Comment:
do we really need a separate function? i see it's used only once

Reply:
i guess since every ep 'creation' function has its own 'destroy' function

Reply:
seems that ucp_ep_destroy_base() is not really the "undo" of ucp_ep_destroy_base()

Reply:
why? it frees all allocated resources in ucp_ep_create_base

Reply:
1. it's not releasing stats
2. why isn't it used from ucp_ep_delete() then?

Reply:
seems like my fault, will check and fix